hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/Compactor.java

```diff
@@ -216,8 +216,9 @@ private FileDetails getFileDetails(Collection<HStoreFile> filesToCompact, boolea
         }
       }
       tmp = fileInfo.get(TIMERANGE_KEY);
-      fd.latestPutTs =
+      long latestPutTs =
         tmp == null ? HConstants.LATEST_TIMESTAMP : TimeRangeTracker.parseFrom(tmp).getMax();
+      fd.latestPutTs = Math.max(fd.latestPutTs, latestPutTs);
```

Comment:
Check  the code, we do not assign any value to fd.latestPutTs, so what is the actual problem here? The latestPutTs info for bulk loaded files can not negative and cause problem?